clang/lib/CIR/CodeGen/CIRGenBuilder.h

```diff
@@ -408,16 +408,25 @@ class CIRGenBuilderTy : public mlir::OpBuilder {
   /// Get a CIR named struct type.
   mlir::cir::StructType getStructTy(llvm::ArrayRef<mlir::Type> members,
                                     llvm::StringRef name, bool body,
-                                    bool packed, const clang::RecordDecl *ast) {
+                                    bool packed, mlir::Attribute ast) {
     const auto nameAttr = getStringAttr(name);
-    std::optional<mlir::cir::ASTRecordDeclAttr> astAttr = std::nullopt;
     auto kind = mlir::cir::StructType::RecordKind::Struct;
     if (ast) {
-      astAttr = getAttr<mlir::cir::ASTRecordDeclAttr>(ast);
-      kind = getRecordKind(ast->getTagKind());
+      if (auto tagDecl = mlir::dyn_cast<mlir::cir::ASTTagDeclInterface>(ast)) {
+        kind = getRecordKind(tagDecl.getTagKind());
+      }
```

Comment:
Please remove curly braces here.

Reply:
fixed in 74c93530a829f58866c5b0914f361cb3e0a53dc6